model/flow/dkg.go

```diff
@@ -31,51 +31,64 @@ func (state DKGEndState) String() string {
+// In the formal cryptographic protocol for a threshold signature with n parties,
+// the individual participants are identified by n public distinct non-negative integers, or simply indices.
+// These public indices are agreed upon by all participants and are used by the low-level
+// Shamir Secret Sharing [SSS].
+// In Flow, the threshold signature keys are generated by a Distributed Key Generation [DKG]. The DKG
+// therefore requires the same SSS indices as an input to generate the private key shares of each participant.
+// Accordingly, the lower-level cryptographic implementation of the threshold signature and DKG
+// works with these indices. The lower-level cryptographic interface requires that the indices are exactly
+// the set {0, 1, ..., n-1}.
+//
 // On the protocol level, only consensus nodes (identified by their nodeIDs) are allowed to contribute
-// random beacon signature shares. Hence, the protocol level needs to map nodeIDs to DKG indices when
+// random beacon signature shares. Hence, the protocol level needs to map nodeIDs to the indices when
 // calling into the lower-level cryptographic primitives.
 //
 // Formal specification:
-//   - DKGIndexMap completely describes the DKG committee. If there were n parties authorized to participate
+//   - DKGIndexMap completely describes the DKG committee. If n parties are authorized to participate
 //     in the DKG, DKGIndexMap must contain exactly n elements, i.e. n = len(DKGIndexMap)
-//   - The values in DKGIndexMap must form the set {0, 1, …, n-1}.
+//   - The values in DKGIndexMap must form the set {0, 1, …, n-1}, as required by the low level cryptography
+//     module.
 //
-// CAUTION: It is important to cleanly differentiate between the consensus committee 𝒞, the random beacon
-// committee ℛ and the DKG committee 𝒟:
+// CAUTION: It is important to cleanly differentiate between the consensus committee 𝒞, the DKG committee 𝒟
+// and the committee ℛ:
 //   - For an epoch, the consensus committee 𝒞 contains all nodes that are authorized to vote for blocks. Authority
 //     to vote (i.e. membership in the consensus committee) is irrevocably granted for an epoch (though, honest nodes
 //     will reject votes and proposals from ejected nodes; nevertheless, ejected nodes formally remain members of
 //     the consensus committee).
-//   - Only consensus nodes are allowed to contribute to the random beacon. We define the random beacon committee ℛ
-//     as the subset of the consensus nodes, which _successfully_ completed the DKG. Hence, ℛ ⊆ 𝒞.
-//   - Lastly, there is the DKG committee 𝒟, which is the set of parties that were authorized to
-//     participate in the DKG. Mathematically, the DKGIndexMap is an injective function
+//   - The DKG committee 𝒟 is the set of parties that were authorized to
+//     participate in the DKG. Mathematically, the DKGIndexMap is an bijective function
 //     DKGIndexMap: 𝒟 ↦ {0,1,…,n-1}.
```

Comment:
curious what you think of the following suggesiton:

```suggestion
//   - The DKG committee 𝒟 is the set of parties that were authorized to participate in the DKG (happy path; or
//     eligible to receive a private key share from an alternative source on the fallback path). Mathematically,
//     the DKGIndexMap is a bijective function DKGIndexMap: 𝒟 ↦ {0,1,…,n-1}.
```

I tried to include a bit more information about the fallback path. So far, I have written "alternative source" ... not sure if maybe writing "trusted dealer" would be possible too? The reason I am not sure is because if we use the result of a prior DKG, I don't know whether this classifies as a "trusted dealer".

Reply:
I agree with your suggestion. A previous DKG isn't a trusted dealer so I prefer the suggested wording